File: Does-my-number-look-big-in-this/Narcissistic.py
def narcissistic(value):
    string = str(value)
    total = 0
    for element in string:
        total += int(element) ** len(string)
    return total == value


print('', narcissistic(153), '\n', narcissistic(
    1652), '\n', narcissistic(122), '\n')


def narcissistic2(value):
    return value == sum(int(element) ** len(str(value)) for element in str(value))


print('', narcissistic2(153), '\n', narcissistic2(
    1652), '\n', narcissistic2(122))

File: Does-my-number-look-big-in-this/Narcissistic.js
function narcissistic(value) {
  let array = value.toString().split("");
  const reducer = (sum, element) => {
    return sum + Math.pow(parseInt(element), array.length);
  };
  return value === array.reduce(reducer, 0);
}
console.log(narcissistic(153), narcissistic(1652), "\n\n");

function narcissistic2(value) {
  let sum = 0;
  const string = value.toString();
  for (let letter of string) {
    sum += Math.pow(parseInt(letter), string.length);
  }
  return sum === value;
}

console.log(narcissistic2(153), narcissistic2(1652));
